### ISSUE TITLE
Document differences from dimensioned

### ISSUE THREAD

**Issue by droundy** [iliekturtles/uom#108]
I presume you looked at dimensioned before writing uom, and it would be convenient to have in the documentation what the differences are between them.

**Comment by iliekturtles** [comment]
Yes, I did look at `dimensioned`. The main difference is that `dimensioned` works in units (meter, foot, ...) where `uom` works in quantities (length, mass, ...). Documentation is my current focus so this is something I'm going to try to get to in the next few weeks. I also recently added #107 to investigate directly supporting units in `uom`.

**Comment by droundy** [comment]
I'm pretty sure I don't understand the distinction. I've certainly teach students to make a distinction between units and dimensions, but I don't see how what your describing differs from dimensioned. Surely you must specify units whenever you are creating a value with some dimension, and as you say the quantities are stored with some base units.

Dimensioned has the idea of quantities with no units but with dimensions as a way to write generic functions that could work with multiple unit systems.

At a quick look it seems that uom is closely tied to the ISQ which seems to closely tie it to SI units, since most other sets of units have different sets of quantities, especially in regard to electromagnetism.  It seems like you couldn't then support gaussian cgs units, atomic units, or most versions of natural units.

**Comment by droundy** [comment]
To explain a bit further about how uom will fail to work with other unit systems than SI, this relates to the number of base units.  Gaussian cgs units have just for four base units counting temperature.  This means that all electromagnetic units after derived, unlike in SI.  There are a few other differences, e.g. the Lorentz force law reads $F = q(E + v/c × B)$ meaning that E and B have the same dimensions, in contrast to SI.  D and H are similarly different from SI.  So in gaussian units, $E^2 + B^2$ is well dimensioned and has dimensions of energy per volume, while in SI units it is an error because you can't add quantities with different dimensions.

**Comment by iliekturtles** [comment]
In regards to dimensions vs. units the idea was that units are only specified at interface boundaries and calculations internal to a library would just work at the dimension level. I'm starting to rethink this (#107) and have some ideas in my head, but haven't written any code recently.

I've never done much work with electromagnetism and never used CGS before so wasn't fully aware of some of the repercussions of all electromagnetic units being derived. I think I'll need to attempt to implement the system using `uom` to experience the short-comings and then see what could be done to possibly overcome them. I just added #109 to address this.
